File: ios/App/App/BatteryPlugin.m
//
//  BatteryPlugin.swift
//  App
//
//  Created by Ivin Antony on 06/03/23.
//
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(BatteryPlugin, "BatteryPlugin",
//           CAP_PLUGIN_METHOD(onBatteryChange, CAPPluginReturnCallback);
           CAP_PLUGIN_METHOD(getBatteryLevel, CAPPluginReturnPromise);
)
//CAP_PLUGIN(MyPlugin, "MyPlugin",
//           CAP_PLUGIN_METHOD(getBatteryLevel, CAPPluginReturnPromise);
//           CAP_PLUGIN_METHOD(startBatteryMonitoring, CAPPluginReturnCallback);
//           CAP_PLUGIN_METHOD(stopBatteryMonitoring, CAPPluginReturnPromise);
//)

    
    
        
     
   
   
  

File: ios/App/App/BatteryPlugin.swift
//
//  BatteryPlugin.swift
//  App
//
//  Created by Ivin Antony on 09/03/23.
//

import Foundation
import Capacitor
import UIKit
import SwiftUI
import Capacitor

@objc(BatteryPlugin)
public class BatteryPlugin: CAPPlugin {

    
    public override func load()  {
        UIDevice.current.isBatteryMonitoringEnabled = true
        
        print("Battery load:")
  
    }
    @objc func batteryLevelDidChange(notification: Notification){
        print("Battery level changed: \(Int(UIDevice.current.batteryLevel * 100))")

        self.bridge?.triggerJSEvent(eventName: "batteryChange", target: "window", data: "{ 'dataKey': 'dataValue' }")
        self.notifyListeners("batteryChange", data: ["dataKey": "dataValue" ])
    }
    @objc func batteryStateDidChange(notification: Notification){
        print("Battery state changed: \(UIDevice.current.batteryState)")
        
        self.bridge?.triggerJSEvent(eventName: "batteryChange", target: "window", data: "{ 'dataKey': 'dataValue' }")
        self.notifyListeners("batteryChange", data: ["dataKey": "dataValue" ])
    
    }
  
    @objc func onBatteryChange(_ call: CAPPluginCall) {
        print("onBatteryChange")
        call.keepAlive = true
        UIDevice.current.isBatteryMonitoringEnabled = true
        
        
        NotificationCenter.default.addObserver(self, selector: #selector(batteryLevelDidChange(notification:)), name: UIDevice.batteryLevelDidChangeNotification, object: nil)
        
        NotificationCenter.default.addObserver(self, selector: #selector(batteryStateDidChange(notification:)), name: UIDevice.batteryStateDidChangeNotification, object: nil)

    }


 
   @objc func getBatteryLevel(_ call: CAPPluginCall) {
       call.keepAlive = true
           call.resolve(["batteryLevel In getBatteryLevel":  (UIDevice.current.batteryLevel * 100)])
    
        }


    
 
}
        
    
    
    
        
     
   
   
  

